Models/IonAuthModel.php

```diff
@@ -0,0 +1,2798 @@
+		                ->get();
+	}
+
+	/**
+	 * @param int|string|array $check_group group(s) to check
+	 * @param int|string|bool  $id          user id
+	 * @param bool             $check_all   check if all groups is present, or any of the groups
+	 *
+	 * @return bool Whether the/all user(s) with the given ID(s) is/are in the given group
+	 * @author Phil Sturgeon
+	 **/
+	public function inGroup($check_group, $id = false, $check_all = false)
+	{
+		$this->triggerEvents('in_group');
+
+		$id || $id = $this->session->get('user_id');
+
+		if (!is_array($check_group))
+		{
+			$check_group = [$check_group];
+		}
+
+		if (isset($this->_cacheUserInGroup[$id]))
+		{
+			$groups_array = $this->_cacheUserInGroup[$id];
+		}
+		else
+		{
+            $users_groups = $this->getUsersGroups($id)->getResult();
+			$groups_array = [];
+			foreach ($users_groups as $group)
+			{
+				$groups_array[$group->id] = $group->name;
+			}
+			$this->_cacheUserInGroup[$id] = $groups_array;
+		}
+		foreach ($check_group as $key => $value)
+		{
+			$groups = (is_numeric($value)) ? array_keys($groups_array) : $groups_array;
+
+			/**
+			 * if !all (default), in_array
+			 * if all, !in_array
+			 */
+			if (in_array($value, $groups) xor $check_all)
+			{
+				/**
+				 * if !all (default), true
+				 * if all, false
+				 */
+				return !$check_all;
+			}
+		}
+
+		/**
+		 * if !all (default), false
+		 * if all, true
+		 */
+		return $check_all;
+	}
+
+	/**
+	 * add_to_group
+	 *
+	 * @param array|int|float|string $group_ids
+	 * @param bool|int|float|string  $user_id
+	 *
+	 * @return int
+	 * @author Ben Edmunds
+	 */
+	public function addToGroup($group_ids, $user_id = false)
+	{
+		$this->triggerEvents('add_to_group');
+
+		// if no id was passed use the current users id
+		$user_id || $user_id = $this->session->get('user_id');
+
+		if(!is_array($group_ids))
+		{
+			$group_ids = [$group_ids];
+		}
+
+		$return = 0;
+
+		// Then insert each into the database
+		foreach ($group_ids as $group_id)
+		{
+			// Cast to float to support bigint data type
+			if ($this->db->table($this->tables['users_groups'])->insert([
+																	$this->join['groups'] => (float)$group_id,
+																	$this->join['users']  => (float)$user_id  ]))
+			{
+				if (isset($this->_cache_groups[$group_id]))
+				{
+					$groupName = $this->_cache_groups[$group_id];
+				}
+				else
+				{
+					$group = $this->group($group_id)->result();
+					$groupName = $group[0]->name;
+					$this->_cache_groups[$group_id] = $groupName;
+				}
+				$this->_cacheUserInGroup[$user_id][$group_id] = $groupName;
+
+				// Return the number of groups added
+				$return++;
+			}
+		}
+
+		return $return;
+	}
+
+	/**
+	 * remove_from_group
+	 *
+	 * @param array|int|float|string|bool $group_ids
+	 * @param int|float|string|bool $user_id
+	 *
+	 * @return bool
+	 * @author Ben Edmunds
+	 */
+	public function removeFromGroup($group_ids = false, $user_id = false)
```

Comment:
Typehint (also need to camelCase these vars)
```suggestion
	public function removeFromGroup($group_ids = false, $user_id = false): bool
```